lib/MetaProcessor/MetaParser.cpp

```diff
@@ -474,6 +474,34 @@ namespace cling {
     return false;
   }
 
+  // dumps/creates a trace of the requested representation.
+  bool MetaParser::istraceCommand() {
+    if (getCurTok().is(tok::ident) &&
+        getCurTok().getIdent().equals("trace")) {
+      consumeToken();
+      skipWhitespace();
+      if (!getCurTok().is(tok::ident))
+          return false;
+      std::string ident = getCurTok().getIdent();
+      consumeToken();
+      skipWhitespace();
+      if (getCurTok().is(tok::eof)) {
+        m_Actions->actOnstatsCommand(ident=="ast" 
+            ? llvm::StringRef("asttree") : llvm::StringRef(ident), llvm::StringRef());
```

Comment:
You can remove the redundant conversion of **ident** to a _std::string_ and then back to a _llvm::StringRef_ starting in MetaParser.cpp at line 485.

I'd also get rid of the name re-mapping, possibly:
```
.stats ast -> Interpreter::dump("ast")
.trace ast -> Interpreter::dump("ast", "*")
.trace ast filter -> Interpreter::dump("ast", "filter")
```